### ISSUE TITLE
Cross compilation of freetype fails

### ISSUE THREAD

**Issue by Cloudef** [NixOS/nixpkgs#160323]
First of all, sorry if this isn't right place for this, but I don't really know where else to ask.

I'm using the following nix file:
```nix
{}:

rec {
   # https://github.com/NixOS/nixpkgs/pull/113122
   tar = fetchTarball"https://github.com/s1341/nixpkgs/archive/android_prebuilt_working.tar.gz";
   pkgs = import tar {};
   android-i686 = import tar {
     crossSystem = {
       config = "i686-unknown-linux-android";
       rustc.config = "i686-linux-android";
       sdkVer = "29";
       ndkVer = "21";
       useAndroidPrebuilt = true;
     };
   };

   test = android-i686.stdenv.mkDerivation {
     name = "test";
     buildInputs = [ android-i686.freetype ];
     buildPhase = ''
       find ${android-i686.freetype.out}
     '';
   };
 }
```

with `nix-build -A test` outputs:

```
error: builder for '/nix/store/zihm0cxjl7p047wjzx6i6mrm49s1xzxz-bash-4.4-p23-i686-unknown-linux-android.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/gr25xalh7dz32zl6haw3636cp3kxhi6a-hook.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-pkg-config-0.29.2-i686-unknown-linux-android.drv-0/pkg-config-0.29.2/glib': Directory not empty
error: 1 dependencies of derivation '/nix/store/2mr2jwhxz56ym4mdv9xlb2x1i6zbc5wb-freetype-2.10.4-i686-unknown-linux-android.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rni4lz32s9igc7l332k13kj6wpx8zm1d-test-i686-unknown-linux-android.drv' failed to build
```

The freetype dependency seems to bring in a hook, that tries to compile bash for the host/target platform. Is there way to avoid this. Am I doing the cross-compilation wrong here? Other dependencies seems to compile fine.

**Comment by ramirez7** [comment]
Just wanted to chime in and say your simple override fixed `freetype` for me (cross-compiling to Windows with `mingwW64`). So thanks a lot! Saved me a lot of time.

--------------

As I researched this, I found that there's really two separate issues with x-compiling `freetype`:

1. `makeWrapper` brings in `bash`. This can be solved with [`makeBinaryWrapper` - a portable, `bash`-free, drop-in replacement.](https://github.com/NixOS/nixpkgs/pull/124556) Something like `freetype = this.freetype.override { makeWrapper = makeBinaryWrapper }` worked for me.
2. `pkg-config` is for some reason using the cross-compiled one instead of the Linux (build machine) one. I found a related issue here: https://github.com/NixOS/nixpkgs/issues/49526 I'm guessing things can be done to improve `freetype`'s use of `pkg-config` when cross-compiled. But I don't exactly understand what `freetype-config` does to begin with :sweat_smile: 

I had a dependency on `freetype` due to `SDL2_ttf`, which seems to use the `freetype-config`. But despite me clobbering the `postInstall` step entirely (which mostly modified `freetype-config`), it compiled perfectly fine. No harm, no foul I guess?
